### DISCUSSION TITLE
Welcome to the V3 transition - please post your issues and questions!

### DISCUSSION THREAD

**Discussion by Bill-Becker** [NREL/REopt-API-Analysis#113] [V3 Transition]
We will be monitoring this discussion forum and will try to be very responsive to posts!

**Comment by jacobjahnke** [comment]
<img width="811" alt="Screenshot 2023-09-26 at 9 19 23 AM" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/98354356/24aacfdd-2ef0-4f45-a2bf-94b00e6e42e2">
There is a required field in Site and Utility when selecting the GHP tech that has no description and the tool will not run.

**Reply by Bill-Becker** [reply]
Hi @jacobjahnke, we are actively working on a fix for this. Thanks for your patience.

**Reply by jacobjahnke** [reply]
Thank you!

**Reply by Bill-Becker** [reply]
Sorry for the delay, GHP has been fixed!

**Reply by jacobjahnke** [reply]
Thank you, it can be run now. But after running I see this screen: 
![Screenshot 2023-10-05 at 9 11 50 AM](https://github.com/NREL/REopt-Analysis-Scripts/assets/98354356/19e152cf-83de-457a-82b0-9dbd0ab30009)

**Reply by jacobjahnke** [reply]
If I look at my 'Saved Evaluations' I can see that it ran but when I click "View Results" of a run I get the screen above too

**Reply by Bill-Becker** [reply]
Can you confirm this is still the behavior your seeing?

**Reply by jacobjahnke** [reply]
I am still seeing the above screen yes. When I try to view results and just make a run.

**Comment by subvip7** [comment]
If you set the "Host effective tax rate (%)" to anything other than the default, it won't run.

**Reply by Bill-Becker** [reply]
This should be fixed now.

**Comment by subvip7** [comment]
Also I get the same problem with changing the "Analysis period (years)" to 20.  It just takes me to the top of the screen with no errors.

**Reply by Bill-Becker** [reply]
Thank you @subvip7, we will be fixing these bugs very shortly.

**Reply by Bill-Becker** [reply]
This should be fixed now.

**Comment by dmbiro** [comment]
Hi!

Im trying to run the REopt for several locations in Alaksa, ranging from Fairbanks to Juneau, and it always gives me the message in the image below. For my last tests I tried the simplest scenario. Just giving the required inputs: location, electricity rate (selected from menu), and building type. I first thought it was something to do with the NSRDB cut-off at lattitude 60, but even when I try locations like Juneau, that is within the NSRDB, I get the same message.

![MicrosoftTeams-image (1)](https://github.com/NREL/REopt-Analysis-Scripts/assets/123515534/88a67103-c9f5-4a4b-b87f-eb4dc732df0f)

**Reply by Bill-Becker** [reply]
I've now observed this issue myself, and we will add this to our list!

**Reply by adfarth** [reply]
@dmbiro this issue should now be resolved!

**Comment by prat777** [comment]
Similar issue with GHP as mentioned in previous responses. All inputs have been filled out, but "Get Results" action scrolls to the top of the page without stating any warnings.

**Reply by Bill-Becker** [reply]
Yes, we are actively working on this, and I will post to this thread when the fix has been deployed.

**Reply by Bill-Becker** [reply]
Sorry for the delay, GHP has been fixed!

**Comment by tommymobley3131** [comment]
Would you be able to add "single family house" as one of the building types? A lot of homeowners and homebuilders are considering adding GSHPs to their home. 

Also, when my teammember tried to use the tool earlier, it did not generate a report for us.

**Reply by Bill-Becker** [reply]
Good suggestion for single family house, but beware that our tool has default values for cost that are more applicable to commercial scale. For residential scale, you'll want to find residential-appropriate cost assumptions for any technology, including GHP. Also, we model vertical bore field configuration ground heat exchangers, where some residential systems use horizontal ground heat exchangers. 

We apologize for GHP not working right now - we are actively working on it and it's been a much tougher challenge than we expected. We should have it working by the end of this week.

**Reply by Bill-Becker** [reply]
Sorry for the delay, GHP has been fixed!

**Comment by tommymobley3131** [comment]
Thanks for the follow up, Bill! This is very helpful and I am so appreciative that you all are working on this.

I’m cc’ing Sajed in case he has any questions about this.

Tommy

Tommy Mobley
Business Development Associate

***@***.***

www.ecosmartsolution.com<http://www.ecosmartsolution.com/>

**Comment by bhhagatsingh** [comment]
Tried to run a GSHP based system with PV and Wind. As someone mentioned before, the screen scrolls to the top but does not provide any results or reports. Already tried running this multiple times since the release, no avail. Kindly fix the issue as soon as possible.

I tried removing the GSHP, CHP options and it ran once - Maybe this will help the devs fix the issue.

**Reply by Bill-Becker** [reply]
Thanks, we plan to have everything fixed by 10/5.

**Reply by Bill-Becker** [reply]
Sorry for the delay, GHP has been fixed!

**Comment by jacobjahnke** [comment]
Can I get an example of what posting a job looks like now? I have code based on the now-seemingly-removed `single_scenario_example.ipynb` which goes as:

`root_url = "https://developer.nrel.gov/api/reopt/stable"`

    return get_api_results(
        post=post,
        API_KEY=REOPT_API_KEY,
        api_url=root_url,
        run_id=None,
    )

**Reply by jacobjahnke** [reply]
I get `Response OK` and `Polling`, but then there are just `KeyError count: 1` and exits with `Breaking polling loop due to KeyError count threshold of 3 exceeded.`

**Reply by Bill-Becker** [reply]
I'd suggest you use the updated [Jupyter Notebook](https://github.com/NREL/REopt-Analysis-Scripts/blob/master/REopt%20API%20Scripts/single_scenario_example.ipynb) for calling V3.

**Reply by jacobjahnke** [reply]
Thank you! I believe there is an error in the `tech_list` when looking at the results, it should be `ElectricStorage` instead of `Storage`

**Reply by Bill-Becker** [reply]
Oh yes good call - feel free to make changes and put up a pull request, or add an issue. Thanks!

**Comment by jamal-54** [comment]
The web tool is not working when I am trying to simulate campus load profiles. However, it is working for a single building load profile.

**Reply by Bill-Becker** [reply]
Thank you, I see that as well. We will work on fixing this. Thanks for your patience.

**Reply by Bill-Becker** [reply]
This should be fixed now. Please let me know if you are still having issues.

**Reply by jamal-54** [reply]
It is working now. Thanks

**Comment by bhhagatsingh** [comment]
Tried simulating a heat load and GSHP. Kept getting this error even when I removed most of my input values.
<img width="937" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/4e850590-8680-4c13-b6e0-af54638b3f91">

**Reply by Bill-Becker** [reply]
Did you also enter a cooling load?

**Reply by bhhagatsingh** [reply]
Yes, I did. I also tried without adding the cooling load and juggled with the electric load as well. None of the iterations worked

**Reply by Bill-Becker** [reply]
Can you explain what you mean by "simulating a heat load"? Did you choose the "Simulate Building" option?

**Reply by prat777** [reply]
I also yield a similar error, but I have defined monthly loads for heating, cooling, and the chiller plant. 
![Screenshot from 2023-10-09 14-12-52](https://github.com/NREL/REopt-Analysis-Scripts/assets/4643760/e9aa0a3c-91de-49a6-bcd0-282ffaeb5234)

**Reply by bhhagatsingh** [reply]
Hey Bill,
I uoloaded the load profile for thermal load and tried running it. Then tried the summed up values for a simulated building . Both ways it didn't work.
<img width="674" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/0b5e3883-54fa-4a85-9083-3b92025bd8df">


For cooling load , I entered the values for the year in tonnes/yr for a single building. I also tried removing it (simulation worked but gave an error)
<img width="854" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/92469b1a-e3b1-4af3-bc70-1c6b6aa8d228">

I think the simulation worked once I changed the location to Fargo from Lund, Sweden . But it gave the same error as dmbiro. 
<img width="448" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/4f9ad000-b92d-4705-9f18-3dfe3b350afd">

**Reply by Bill-Becker** [reply]
We identified that the issue is within the web tool UI for loading the results page when a user-uploaded file was used. We should have a fix for this early this week. We've also identified and are working on an issue with monthly heating load input. Thanks for your patience.

**Reply by Bill-Becker** [reply]
We believe we have a workaround fix for this, ready for you to test if it's working for you. You may not be able to see the file name of the load profile that you uploaded, but the data will be loaded properly.

**Comment by dmbiro** [comment]
When setting boundaries to sizing PV that are outside of the optimal solution I get the error below. Lets say the initial solution found by REopt in this scenario was suggesting 200 kWdc, but I know I want to install more PV for this specific site (minimum 210 kWdc), and it doesnt let me adjust that.  

![image](https://github.com/NREL/REopt-Analysis-Scripts/assets/123515534/407fb593-7627-46fe-83fa-2eb7a118a13e)

**Reply by Bill-Becker** [reply]
I tried to replicate this issue and couldn't. I'm suspecting that it could be something about the PV size relative to the load, e.g. the larger PV production makes the net load go negative. That should be fine, PV would curtail. But let me know if you think that is the scenario or provide any other details to replicate this issue.

**Comment by dmbiro** [comment]
When I download the results PDF, the section that should display the results and inputs isn't expanded; I can only see the title of each section, as shown in the image below:

![image](https://github.com/NREL/REopt-Analysis-Scripts/assets/123515534/dc30e3b4-d1e9-4f70-bf59-fe82dd3fef0a)

**Comment by bhhagatsingh** [comment]
Update on errors : I tried running the reOPT tool again on the 10th Oct, and it hadn't worked for simulating a combined electric and thermal load with daily values uploaded. Tried again today, not much improvement. However, I found this on the saved evaluations page : 
<img width="971" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/f69b2413-4cc0-4932-9027-08721e014e71">

I tried opening these evaluations by pressing on 'View results'. I got this error : 
<img width="775" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/736c9a25-008c-4b23-b4a7-96c65636f90e">

I also tried to compare the two simulations. I got this page, might help the team fix the problem.
<img width="741" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/4d48ce4e-82ca-4c66-a2fa-9f4acf115ae9">

It would be helpful if you can analyse these individual simulations and find out the problem.

**Reply by Bill-Becker** [reply]
Thank you for your patience with this and describing the problem to us. We have identified the issue with viewing the results page for runs which have uploaded loads (uploading a file of 8760) - please confirm that is what you are using to upload the load? If so, we are actively working to fix this in the web UI.

**Reply by bhhagatsingh** [reply]
Yes, I uploaded files with 8760 load values(for both electric and thermal) for running the simulations.

**Reply by Bill-Becker** [reply]
We believe we have a workaround fix for this, ready for you to test if it's working for you. You may not be able to see the file name of the load profile that you uploaded, but the data will be loaded properly.

**Comment by astronobri** [comment]
When I try to run results, the page scrolls to the top with no error message. This happened after I modified the PV inputs and selected "existing PV system" then specified the Existing Generator Size (kW) as 45000. Changing the generator size still gave the same result.

**Reply by Bill-Becker** [reply]
Thank you for letting us know. I've replicated the issue, and we will work on fixing this.

**Reply by Bill-Becker** [reply]
I believe this is fixed as of this morning, but please let us know if you still have any issues.

**Reply by astronobri** [reply]
This seems to be working now, however the output recommended PV size is the new PV + existing PV, but shouldn't it be minus the existing PV? For example, without adding existing solar REopt recommends building 62.5 MW. When I add in the 45 MW existing solar, the new recommended build becomes 107.5 MW.

**Reply by Bill-Becker** [reply]
This might be more of a nomenclature issue. We say the total recommended installed PV capacity at the top, and that includes both existing and new. I can see how this is a little misleading, but we explain the "existing+new" directly under that value. Let me know if something seems wrong, other than that.

**Reply by astronobri** [reply]
That makes sense, however I'm still not sure why REopt gives different total installation sizes for both scenarios (when including 45 MW existing PV, and when not including any existing PV). In the case without existing PV, the total size is 62 MW. In the case with adding 45 MW of existing PV, the total size then became 107 MW. I made sure that in the run with existing PV I selected "gross load" so in theory REopt should have recommended a total installation size of 62 MW (45 MW existing + 17 MW new) and not 107 MW (45 MW existing + 65 MW new), if that makes sense? I'll attach screenshots of the two runs.

![Screenshot 2023-12-04 at 11 40 21 AM](https://github.com/NREL/REopt-Analysis-Scripts/assets/29382951/191864f7-f7da-42d7-8714-3810eb6a5692)
![Screenshot 2023-12-04 at 11 41 58 AM](https://github.com/NREL/REopt-Analysis-Scripts/assets/29382951/22ca00e6-9507-47c9-b303-132ee721cfab)

**Reply by astronobri** [reply]
I also tried switching from "gross load" to "net load" when I added in the existing PV, but the results were still the same (107 MW total). Shouldn't the results be different?

**Comment by brtietz** [comment]
Just a quick note that https://github.com/NREL/REopt-Analysis-Scripts/wiki/2.-Run-analyses-using-the-API is out of date. "status" has been moved to a top level variable and is no longer nested under "Scenario"

**Reply by Bill-Becker** [reply]
Good catch Brian, thanks. @adfarth can you update this?

**Reply by adfarth** [reply]
Updated that a few other out of date items. Please continue to let us know if you find incorrect documentation--thank you!

**Comment by subvip7** [comment]
After the V3 upgrade, I was able to see my newly created evaluations under the "Saved Single Site Evaluations" section.  Now even if I create a new one it doesn't show here (the old, old ones from before the update are still there under "Saved Single Site Evaluations Before 9/25/2023").  Even if I create a new one, it no longer is accessible.  Here's one I had before that is gone: cf7f3ba2-db4e-4567-909e-4f6d4ccf5c39.  
![image](https://github.com/NREL/REopt-Analysis-Scripts/assets/3869493/731bc541-7c2b-4cee-81ee-5850b1b1482a)

Any help would be appreciated.  Thanks!

**Reply by Bill-Becker** [reply]
Hmm, this is strange. I have not seen this behavior. Are you sure you are using the same login? We did recently update so that you could access evaluations prior to September, but for me it shows my older and newer evaluations. Is that id cf7f3ba2-db4e-4567-909e-4f6d4ccf5c39 the ID that is in the web tool URL or is that the run_uuid (if you know)?

**Reply by subvip7** [reply]
I tried it again completely from scratch and the problem remains.  It simply never shows any "Saved Single Site Evaluations" even if I just created it under my usual account.  This time I saved the URL: https://reopt.nrel.gov/tool/results/2e1f5dcd-75e7-4ed5-8518-2c0bfc711710.  The big challenge is that I frequently make small changes to each of my simulations and without it being there, I have to put all my values in from scratch.  Any help would be appreciated.  Thanks.

**Reply by subvip7** [reply]
Also, I also tried just a super simple run with a location in the US, and no default values changed.  It also doesn't show up in my saved evaluations.  Here's that one https://reopt.nrel.gov/tool/results/be0624ff-c9ef-44b3-a8e1-f72b91ffd36e.

**Reply by subvip7** [reply]
I did one last experiment and that was to create a new account with reOpt and try the test above.  I used the same settings as above and it worked.  So the problem must be my other email account is just not allowing new evaluations to be saved.

**Reply by Bill-Becker** [reply]
Were you signing in with a gmail account? Would you like me to attempt to get your original account working? If so, please provide the email that you were using.

**Reply by subvip7** [reply]
Sure.  My gmail address that was used was joshuashinar@gmail.com

**Comment by bhhagatsingh** [comment]
Tried to run ReOPT recently, found this error : 
<img width="907" alt="image" src="https://github.com/NREL/REopt-Analysis-Scripts/assets/146493655/7d39cecb-886e-4925-946f-aa57282e0d96">

Any idea how to fix?

**Reply by Bill-Becker** [reply]
Sorry about that. The issue you’re having is because every REopt.jl user is currently using the same NREL api_key, which is currently hard-coded in REopt.jl and needs to be corrected! This is leading to the max 1000 hits per day for the Wind Toolkit, and/or 1000 per hour for PVWatts. 

I need to fix this so the REopt.jl user enters their own API key, either by setting a Julia environment variable or as an argument to the run_reopt function call (definitely leaning toward env variable based on reading up on best practices). This will require that additional line of code in the workflow to run REopt. I will work on this this week. In the meantime, if you are using a feature branch of REopt.jl, you can put your own API key in [this location](https://github.com/NREL/REopt.jl/blob/0085341139a2999eb14b486680832014fd5c1f3f/src/keys.jl#L31) and update the branch in your environment.

**Reply by Bill-Becker** [reply]
This issue should be resolved. Please let me know if you run into this again.

**Reply by danWagar** [reply]
Not sure if this is the correct place for this, please redirect me if not.

I ran into this issue today via the https://developer.nrel.gov/api/reopt/v3/job endpoint. Is there any chance this issue is being worked on or may be resolved in the near future? Trying to determine if it is viable to switch to v3 or if I should stick with v2 for the time being.

**Reply by bhhagatsingh** [reply]
This got resolved @danWagar . The simulations ran smoothly after a recent fix on v3.

**Reply by danWagar** [reply]
I'm not using the reopt webtool or running the branch in my development environment, I am using the API. I reviewed the instructions for setting the environment variable but that does not apply in this situation. Do you know specifically if there is any documentation on how to provide the developer api key when using the api?

**Reply by danWagar** [reply]
Here is my error output from a job ran yesterday:


```
"messages": {
    "errors": {
      "core_utils.jl_413": [
        " Error occurred when calling PVWatts HTTP.Exceptions.StatusError(429",
        " GET",
        " apipvwattsv8.jsonapi_keygAfosXcQ9Ldfw3qXqvKVb7PxMEkYigozmC9R3mXQlat33.135318lon91.961597tilt10.0system_capacity1azimuth180.0module_type1array_type1losses14.0dc_ac_ratio1.2gcr0.4inv_eff96.0timeframehourlydatasetnsrdbradius0",
        " HTTP.Messages.ResponsennHTTP1.1 429 Too Many RequestsrnDate Tue",
        " 28 Nov 2023 174053 GMTrnContentType applicationjsonrnTransferEncoding chunkedrnConnection keepalivernAccessControlAllowOrigin rnRetryAfter 2476rnStrictTransportSecurity maxage31536000 preloadrnXApiUmbrellaRequestId cc2s2g8jvp0mjt01ut70rnXCache MISSrnXRatelimitLimit 1000rnXRatelimitRemaining 0rnXVcapRequestId de7e32a752be41b168a9e81cf2d763d1rnXFrameOptions DENYrnXContentTypeOptions nosniffrnXXSSProtection 1 modeblockrnrnn  error n    code OVER_RATE_LIMIT",
        "n    message You have exceeded your rate limit. Try again later or contact us at httpsdeveloper.nrel.govcontact for assistancen  n)",
        " Error occurred when calling PVWatts HTTP.Exceptions.StatusError(429",
        " GET",
        " apipvwattsv8.jsonapi_keygAfosXcQ9Ldfw3qXqvKVb7PxMEkYigozmC9R3mXQlat35.156932lon89.961975tilt10.0system_capacity1azimuth180.0module_type1array_type1losses14.0dc_ac_ratio1.2gcr0.4inv_eff96.0timeframehourlydatasetnsrdbradius0",
        " HTTP.Messages.ResponsennHTTP1.1 429 Too Many RequestsrnDate Tue",
        " 28 Nov 2023 174055 GMTrnContentType applicationjsonrnTransferEncoding chunkedrnConnection keepalivernAccessControlAllowOrigin rnRetryAfter 2481rnStrictTransportSecurity maxage31536000 preloadrnXApiUmbrellaRequestId cc2s2gn255od8csb3cg0rnXCache MISSrnXRatelimitLimit 1000rnXRatelimitRemaining 0rnXVcapRequestId 5d837c74c4d646056e2e0b858dba6b33rnXFrameOptions DENYrnXContentTypeOptions nosniffrnXXSSProtection 1 modeblockrnrnn  error n    code OVER_RATE_LIMIT",
        "n    message You have exceeded your rate limit. Try again later or contact us at httpsdeveloper.nrel.govcontact for assistancen  n)"
      ]
    },
    "warnings": {

    },
    "has_stacktrace": false
  }

```
